### PR TITLE
chore: deprecate the  @aws-cdk/aws-cloudformation module

### PR DESCRIPTION
**Pull request by skinny85**
This is only for the v2-main branch,
as this module is used in the public API of two stable modules,
`@aws-cdk/aws-codepipeline-actions` and `@aws-cdk/custom-resources`.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by skinny85** [comment]
@rix0rrr I assume this will fail on the backwards compatibility check. Any idea what to do with it? Should I just add these violations to `allowed-breaking-changes.txt`, or do we need something smarter here?

File: packages/@aws-cdk/aws-codepipeline-actions/lib/cloudformation/pipeline-actions.ts [hunk 1]
```diff
@@ -166,7 +165,7 @@ interface CloudFormationDeployActionProps extends CloudFormationActionProps {
    * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/using-iam-template.html#using-iam-capabilities
    * @default None, unless `adminPermissions` is true
    */
-  readonly capabilities?: cloudformation.CloudFormationCapabilities[];
+  readonly capabilities?: cdk.CfnCapabilities[];
```
Comment (nija-at) [inline packages/@aws-cdk/aws-codepipeline-actions/lib/cloudformation/pipeline-actions.ts:168]: We should do this the usual way we change APIs. Mark it as `@deprecated` on master, and still support the old APIs. As part of the deprecated APIs work, these will be stripped off in v2.

**Review by nija-at — CHANGES_REQUESTED** [review]

File: packages/@aws-cdk/aws-codepipeline-actions/lib/cloudformation/pipeline-actions.ts (hunk 1 above)
Comment (skinny85) [reply packages/@aws-cdk/aws-codepipeline-actions/lib/cloudformation/pipeline-actions.ts:168]: We can do that, but that will not achieve the desired outcome here.

If you leave this API in `@aws-cdk/aws-codepipeline-actions`, this means you need to leave the `@aws-cdk/aws-cloudformation` module inside of `aws-cdk-lib`, which means you cannot deprecate it.

So this entire PR would effectively be discarded.

Comment (nija-at) [reply packages/@aws-cdk/aws-codepipeline-actions/lib/cloudformation/pipeline-actions.ts:168]: We could keep it in there, until deprecated APIs are stripped off. Once we achieve that and it's in place, we can special case the removal of this package. 

Actually, if all APIs on the `aws-cloudformation` package are deprecated, then none of its APIs will be visible in `aws-cdk-lib`, so no harm in still keeping it in.

However, it's important that all usages of the cloudformation packages be deprecated for this to work.

Comment (skinny85) [reply packages/@aws-cdk/aws-codepipeline-actions/lib/cloudformation/pipeline-actions.ts:168]: OK. Let's not keep this for now then.

Comment (skinny85) [reply packages/@aws-cdk/aws-codepipeline-actions/lib/cloudformation/pipeline-actions.ts:168]: Moved to #12670.

**Pull request closed without merging**